### ISSUE TITLE
No features even with fdr_level=0.99

### ISSUE THREAD

**Issue by AMilkov** [blue-yonder/tsfresh#137]
For some reason no features are significant enough.

Here is the notebook export.
[tsfresh.zip](https://github.com/blue-yonder/tsfresh/files/691293/tsfresh.zip)

Part of the feature significance test:
[test_feature_significance.zip](https://github.com/blue-yonder/tsfresh/files/691295/test_feature_significance.zip)

Is it normal workflow to assume that all the features are binary and write report when this is not true?

**Comment by MaxBenChrist** [comment]
Is your index right? You do something like `ind = df[df.id.isin(ids)].index.get_values()` to extract a label based index and then `df.iloc[ind, :]`. That should be `df.loc[ind, :]`, shouldn't it? Not sure if this is the reason here.

What you can do is to extract all features and then do a filtering from sklearn http://scikit-learn.org/stable/modules/feature_selection.html. 

Also if you look at the report, most of your features are constant (=non significant) or only have two values:

E.g. this line here
```
[target_binary_feature_binary_test] A binary feature should have only values 1 and 0 (incl. True and False). Instead found {0.34939947477545796, 0.0097505415738271296} in feature ''fundamental_49__cwt_coefficients__widths_(2, 5, 10, 20)__coeff_9__w_10''.
```

So the feature column `fundamental_49__cwt_coefficients__widths_(2, 5, 10, 20)__coeff_9__w_10`
 only contained the values 0.34939947477545796 and 0.0097505415738271296, which is bad. There is not enough variance in your time series.

**Comment by MaxBenChrist** [comment]
What kind of data is this?

**Comment by AMilkov** [comment]
@MaxBenChrist 

The data is from this competition: https://www.kaggle.com/c/two-sigma-financial-modeling/data

I was thinking that the best possible validation for new feature extraction tool will be to show it's usefulness in a Kaggle competition. 

In this case (since the DataFrame index is numerical and is single level) there is no difference between `df.iloc[ind, :]` and `df.loc[ind, :]` - the underlying data and the end result are exactly the same.

I understand that some of the features do not have enough variance and are not useful - this is OK, however all 108 of them to be utter garbage is kind of unexpected.

I can not give you the data, since there is a T&C agreement involved, but you can easily obtain it directly from Kaggle.

Attached is the result of
```
df.loc[ind, 'fundamental_49'].value_counts()
```
Note that **fundamental_49** has 101 different values.
[fundamental_49.zip](https://github.com/blue-yonder/tsfresh/files/691913/fundamental_49.zip)

**Comment by MaxBenChrist** [comment]
We had the same idea and I also wanted to look into that competition. Have you tried to turn off the filtering and use one of the sklearn methods instead?

**Comment by MaxBenChrist** [comment]
@AMilkov : I just had another look at your notebook. You are fitering the features for only 3 ids `ids = [10, 11, 12]`. then it is clear why the filtering procedure is rejecting. Our developed filtering method needs at least 30-40 samples to work properly. 

It is a method that is based on statistical methods that control the false extraction rate. If you do not have enough samples the probabilities that differences in the distributions of the features occur  due to statistical noise are too high and the filtering method will kick out all features. Try to run the filtering procedure with more samples, I am sure it will select some features.

**Comment by MaxBenChrist** [comment]
I am closing this, if the filtering procedure is still not selecting features for 30-50+ samples, please reopen that issue. 

Otherwise good luck with your kaggle endeavors :)

**Comment by AMilkov** [comment]
@MaxBenChrist 
I did the same significant feature selection against 30 IDs and after 1h22m the result was the same - not even single feature was selected.  
I am not sure if this is a bug or intended , so it is up to you to investigate it or not.

So far I do not have comparison against sklearn feature selection algorithms, but will keep you informed once I have that.

**Comment by MaxBenChrist** [comment]
Okay, if you find that other feature selection algorithms are selecting features I will look into this problem!

**Comment by MaxBenChrist** [comment]
You mentioned 1h22m to process 25% of the data, you can try to reduce that time with ReasonableFeatureExtraction Settings.

**Comment by tchekov-** [comment]
I ran into the same problem when trying to select features with too few examples.

I think it would be good to mention this behaviour in the documentation, maybe more people will try tsfresh with a small dataset first.

**Comment by mrektor** [comment]
I have the same problem.
I have three different dataset, each containing different timeseries. I have 80 samples per dataset and my task is binary classification.

The feature extraction took a long time to be computed, but eventually it outputed a df of shape (80, 17468).

Now the problem is that when i try to select important features, `select_features` always return me an empty dataframe.

Now, while I cannot be sure that these datasets can be appropriately classified, my intuition tell me that i should expect at least some features to stand out for extreme fdr values.
I played with `fdr_level` with extremely low values (and with extremely high, just to double check), but the routine always output me an empty dataframe for each dataset.

Is this behaviour expected?

**Comment by NoisySignal** [comment]
@MaxBenChrist you wrote above "Our developed filtering method needs at least 30-40 samples to work properly".

Where is this mentioned in the documentation or associated paper or report?

If the number of samples is an important requirement then it should be made obvious that this is the case. The alternative is to let the user try and find this out by painful and frustrating trial and error.
